metricbeat/module/zookeeper/mntr/data.go

```diff
@@ -36,9 +36,9 @@ var (
 	schema       = s.Schema{
 		"version": c.Str("zk_version"),
 		"latency": s.Object{
-			"avg": c.Int("zk_avg_latency"),
-			"min": c.Int("zk_min_latency"),
-			"max": c.Int("zk_max_latency"),
+			"avg": c.Float("zk_avg_latency"),
```

Comment:
@ruflin This is a breaking change for zookeeper integration, as [integration](https://github.com/elastic/integrations/blob/main/packages/zookeeper/data_stream/mntr/fields/fields.yml) expects `long`. I don't think it's safe to push it.

Reply:
Yeah, I wanted to look at this if it's a permanent bug or a version mismatch

Reply:
Oh yes, it's a version mismatch from zookeeper 3.5 to 3.7. We'll need to upgrade the module and add the new version to the testing image in the CI

Reply:
Remember to update the `fields.yml` file too.

If this change is breaking, we should duplicate the field instead of modifying it.

Reply:
We cannot "just" duplicate the field because those keys must be unique. So instead it should be something like `avg_f`. Does that work for you @pfcoperez ?

The problem is the mapping, because we are producing a JSON output and JSON specification has no notion of integer numeric values anyways.

@jsoriano @mtojek have you ever tried to add 2 mapping types for the same metricbeat field? I know we can do it at elasticsearch level because we can set a field like `keyword` and `text` at the same time.

Reply:
> We cannot "just" duplicate the field because those keys must be unique.

Yes, this is what I meant, sorry :slightly_smiling_face: 

> have you ever tried to add 2 mapping types for the same metricbeat field?

I don't think this is possible, a field in an elasticsearch index can have only one mapping.

> I know we can do it at elasticsearch level because we can set a field like `keyword` and `text` at the same time.

I guess you refer to "[multi-fields](https://www.elastic.co/guide/en/elasticsearch/reference/7.16/multi-fields.html)". We could try to add a `float` field inside the current `long`, but this may be a bit hacky, the purpouse of this feature is different to what we need here. To access the float value you would access something like `...avg.float`.

Reply:
Oh yeah, I was refering to multi-fields indeed. yeah so we are in the same crossroad.

So let's wait to see @pfcoperez response to what works better for them in terms of field names and I'll update the PR myself.

Reply:
I think we can keep the field as float. `0` will be parsed as `0.0`. In fact `ZK 3.7` sometimes sends ingeger formated values. e.g: `3`.

So, in a nutshell, if we changed these fields to `Float` we would parse ZK 3.6+ and previous responses alike, right?

Reply:
Trying to understand on my end why the change from Int to Float is a breaking change?

Reply:
Thanks for checking @ruflin , the breaking change happens at ZooKeeper side in relation to the beats module.
Currently, this module tries and fails to parse these metrics as integers. Until 3.5, zookeeper provided only integer values for the latency metrics.

However, since 3.7, it occasionally (whole values are not represented as `X.0`) responds with float number strings.

Take this example:

- ZK 3.5:

```
zk_avg_latency	0
zk_max_latency	29
zk_min_latency	0
```

- ZK 3.7:

```
zk_avg_latency	6.1223
zk_min_latency	0
zk_max_latency	73
```

I've checked and it seems that the beats parser works correctly when it tries to parse a whole number string (e.g `42` ) as a `Float`: `c.Float("zk_avg_latency"),` of course, it also succeeds in parsing float number strings (e.g `42.3`) so changing these fields to `Float` will make the module compatible with all ZooKeeper versions.

Reply:
> Trying to understand on my end why the change from Int to Float is a breaking change?

@ruflin it could be breaking on ingestion because it is a change on field types (from long to float), it could be also breaking if something as a dashboard is consuming these values as long, and they start being stored as floats. But I think we are safe here as floats can be ingested into long fields. If this value is used in dashboards, it will probably also work with floats. 

We can also play safer, and don't do the change in the mapping, then these values will be collected, but only its integer part would be ingested.

To store them as the floats they are, we need to change the data types in `metricbeat/module/zookeeper/mntr/_meta/fields.yml` from long to float.

The question would be if we only want this to "work", then the change to `Float` here is enough and it probably doesn't produce breaking changes, or if we want to also index it as proper `float`, then we need to change the mapping, and we need to test that nothing breaks. If something breaks, we need to do something as storing the new value in a new field.

Reply:
> How valuable are the digits after the comma?

For Elastic cloud they are not super valuable, we can live without them. However having them is nice because visualizations quality will improve. For example, a `0.0` latency is extremely improbable and it might be interesting to discriminate between that situation and `0.XY` (more probable cases).

Anyway, So far we have lived  with just the integer part (as ZK didn't went further) and that's OK. A different thing is: How valuable are these fractional digits to other users? I don't know.

> We can also play safer, and don't do the change in the mapping, then these values will be collected, but only its integer part would be ingested.

With the above mentioned, if this is safe, this can be a nice sweet spot to fix Cloud monitoring issues and if any of our users need the decimal digits we can always follow up in future releases.

Reply:
Ok, then I think that we can go on with the changes here, so we at least support the newer versions, and we can consider in a follow up to properly index the floats.